### ISSUE TITLE
Selector tool

### ISSUE THREAD

**Issue by chriseth** [powdr-labs/powdr#1895]
We should add a selector tool to more easily write and understand "match-like" constructs in powdr:

```
pc' = XXIsZero * l + (1 - XXIsZero) * (pc + 1)
```

should be replaced by
```
pc' = select_on(XXIsZero, [(0, l), (1, pc + 1)])
```

A similar example for boolean columns can be found in https://github.com/powdr-labs/powdr/issues/1894 .

We could also add an "else" case:
```
pc' = select_on_else(XXIsZero, [(0, l), (1, pc + 1)], 0)
```

**Comment by Schaeff** [comment]
Could this be a match expression?
```rust
pc' = match XXIsZero {
   0 => l,
   1 => pc + 1,
   _ => panic("XXIsZero should have been boolean")
}
```

**Comment by chriseth** [comment]
Yes, but then we have to explain how to turn match expressions to algebraic expressions. I think this is something we considered a long time ago. We could implement it like a user-implemented operator. But I'm not sure if this is too confusing because it behaves differently depending on if it is used in algebraic expression context or elsewhere.
